cdap-cli/src/main/java/co/cask/cdap/cli/command/AbstractSetPreferencesCommand.java

```diff
@@ -73,4 +74,57 @@ protected void setPreferences(String[] programIdParts, PrintStream printStream,
         throw new IllegalArgumentException("Unrecognized element type for preferences " + type.getTitleName());
     }
   }
+
+  protected String determinePattern(String action) {
+
+    switch (action) {
+      case "set":
```

Comment:
Also looks like we can combine the two switch statements into one and use the 'action' string to form the format.